src/app/yourcity/factoryclass.ts

```diff
@@ -0,0 +1,25 @@
+
+
+export class BackgroundFactory {
+  private timeOfDay: string;
+  private BlackOrWhite: string;
+
+  constructor(timeOfDay: string, BlackOrWhite: string) {
```

Comment:
type not a string -> Enum and rename to  `constructor(periodOfDay: PeriodOfDay....`